music.js

```diff
@@ -1,17 +1,29 @@
-// Define the Artist constructor function here, with a 'name' property that can be set at instantiation,
+// Define the Artist constructor function here, with a 'name' property that can be
+//set at instantiation,
 // and a 'songs' property that is an empty array to start
-function Artist() {
-
+function Artist(name) {
+  this.name = name;
+  this.songs=[];
 }
 
-// Define the Song constructor function here, with 'title' and 'artist' properties that can be set at instantiation,
+// Define the Song constructor function here,
+//with 'title' and 'artist' properties that can be set at instantiation,
 // and a 'playCount' property that is an empty array to start
-function Song() {
-
-  // When a new song is instantiated, you'll want to add that song to its artist's 'songs' array.
+function Song(title,artist) {
+  this.title = title;
+  this.artist = artist;
+  this.playCount = [];
+  // When a new song is instantiated, you'll want to add that
+  //song to its artist's 'songs' array.
   // You'll probably want to do that in here somewhere...
+   Artist.prototype.add = function(title){
+   this.songs.push(title);
+}
```

Comment:
This function is being defined _each_ time you create a new song – those operations aren't necessary. Move this function definition outside of the Song constructor function.